spec.html

```diff
@@ -122,6 +122,32 @@ <h1>
 		</emu-note>
 	</emu-clause>
 
+	<emu-clause id="sec-create-type-error-copy" type="abstract operation">
+		<h1>
+			CreateTypeErrorCopy (
+				_realmRecord_: a Realm Record,
+				_originalError_: an ECMAScript language value
+			)
+		</h1>
+		<emu-alg>
+			1. Let _newError_ be a newly created *TypeError* object.
+			1. NOTE: _newError_ is created in _realmRecord_.
+			1. If Type(_originalError_) is Object, then
+				1. Let _message_ be Completion(Get(_originalError_, "message")).
+				1. If _message_ is not an abrupt completion, then
+					1. Let _msgString_ be Completion(ToString(_message_)).
+					1. If _msgString_ is not an abrupt completion, then
+						1. NOTE: Copying the string message when possible.
+						1. Perform CreateNonEnumerableDataPropertyOrThrow(_newError_, "message", _msgString_).
+				1. Let _cause_ be Completion(Get(_originalError_, "cause")).
```

Comment:
I'm worrying if this closes how we gonna improve this in the future, for example, by structured-clone.

Reply:
As the error "cause" property can be arbitrary values, future improvements are still possible as it doesn't break the assumption. Would you mind elaborating on how this would prevent the improvements?

Reply:
Improving it will bring observable changes. if someone relies on it, it will break

Reply:
I'd suggest that, if we make a ShadowRealm which does structured clone, that it be constructed differently with that opt-in. So I don't think we need to worry about this forward compatibility issue.